Source/Dafny/Compilers/Compiler-Csharp.cs

```diff
@@ -484,25 +546,55 @@ public CsharpCompiler(ErrorReporter reporter)
       }
     }
 
+    public override bool NeedsCustomReceiver(MemberDecl member) {
+      Contract.Requires(member != null);
+      //Dafny and C# have different ideas about variance, so not every datatype member can be in the interface.
+      if (!member.IsStatic && (member.EnclosingClass is IndDatatypeDecl || member.EnclosingClass is CoDatatypeDecl)) {
+        DatatypeDecl d = member.EnclosingClass is IndDatatypeDecl ind ? ind : member.EnclosingClass as CoDatatypeDecl;
+        foreach (var tp in d.TypeArgs) {
+          Predicate<Type> InvalidType = ty => ty.AsTypeParameter != null && ty.AsTypeParameter.Equals(tp)
+                                              || ty.TypeArgs.Exists(a => a.AsTypeParameter.Equals(tp));
+          Predicate<Formal> InvalidFormal = f => !f.IsGhost && InvalidType(f.SyntacticType);
+          switch (tp.Variance) {
+            //Can only be in output
+            case TypeParameter.TPVariance.Co:
+              if (member is Function f && f.Formals.Exists(InvalidFormal)
+                  || member is Method m && m.Ins.Exists(InvalidFormal)) return true;
```

Comment:
Put the `if` branch (that is, `return true;`) in curly braces. Same on L568 (and everywhere).